File: Code/Components/DanpheEMR.ServerModel/MedicalRecords/FinalDiagnosisModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class FinalDiagnosisModel
    {
        [Key]
        public int FinalDiagnosisId { get; set; }
        public int PatientVisitId { get; set; }
        public int PatientId { get; set; }
        public int ICD10ID { get; set; }

        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }

        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public bool IsActive { get; set; }


        public override int GetHashCode()
        {
            return this.ICD10ID.GetHashCode();
        }
        public override bool Equals(object obj)
        {
            if (!(obj is FinalDiagnosisModel))
                throw new ArgumentException("Obj is not an FinalDiagnosisModel");
            var icd = obj as FinalDiagnosisModel;
            if (icd == null)
                return false;
            return this.ICD10ID.Equals(icd.ICD10ID);
        }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/medical-records/outpatient-list/final-diagnosis/final-diagnosis.model.ts
import * as moment from "moment";

export class FinalDiagnosisModel {
    public FinalDiagnosisId: number = 0;
    public PatientId: number = 0;
    public PatientVisitId: number = 0;
    public ICD10ID: number = 0;
    // public CreatedBy: number = 1;
    // public CreatedOn: string = moment().format("YYYY-MM-DD");
    public IsActive: boolean = true;
}